contrib/add-capsule-header.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+
+import sys
+import uuid
+import argparse
+import struct
+
+CAPSULE_FLAGS_PERSIST_ACROSS_RESET = 0x00010000
+CAPSULE_FLAGS_POPULATE_SYSTEM_TABLE = 0x00020000
+CAPSULE_FLAGS_INITIATE_RESET = 0x00040000
+
+def main(args):
+
+    # parse GUID from command line
+    try:
+        guid = uuid.UUID(args.guid)
+    except ValueError as e:
+        print(e)
+        return 1
+    try:
+        with open(args.bin, 'rb') as f:
+            bin_data = f.read()
+    except FileNotFoundError as e:
+        print(e)
+        return 1
+
+    # check if already has header
+    hdrsz = struct.calcsize('<16sIII')
```

Comment:
Doesn't this fall apart when you have a header that is bigger than 28 bytes already there?  Or no header there?

Reply:
No, I've been testing it on a 4k Dell image and a 0k Lenovo image.

Reply:
And on the Lenovo case it doesn't just replace the front of the file?   I guess if you had a properly crafted hdr[3] it might, but your heuristic is probably enough.

Reply:
In the Lenovo case it just writes the new header and then the entire payload appended. I figured the imagesz heuristic was going to be good enough in reality.